services/filesstore/s3store.go

```diff
@@ -142,11 +142,8 @@ func (b *S3FileBackend) TestConnection() error {
 	if exists {
 		mlog.Debug("Connection to S3 or minio is good. Bucket exists.")
 	} else {
-		mlog.Warn("Bucket specified does not exist. Attempting to create...")
-		err := b.client.MakeBucket(context.Background(), b.bucket, s3.MakeBucketOptions{Region: b.region})
-		if err != nil {
-			return errors.Wrap(err, "unable to create the s3 bucket")
-		}
+		mlog.Warn("Bucket specified does not exist.")
```

Comment:
This changes the behavior at a global level than just for system managers. The testConnection also happens when the server boots up, and it creates a bucket if that does not exist. That is intentional behavior and the permissions are controlled by the s3 user.

I think what we want here is to not create the bucket _only if_ the user is not a sysadmin or it's not made via an API call.

Reply:
If that's the case, then I think we should be explicitly creating a bucket (e.g. via a CreateBucket method) during startup. A function called "TestConnection" should really just be testing a connection. Similarly, an API endpoint titled "Test AWS S3 connection" (according to our API reference) should only be testing a connection. If we need to create a new bucket, we should explicitly state so. Our API reference does not mention the fact that a new bucket will be created if it does not exist.

If a system admin wants to use a new S3 bucket via the REST API, they can do so via the `/config/patch` endpoint.

Reply:
I like that.

We can let TestConnection return the status of the exists variable (or maybe have an `ErrNoBucket` error). And then on startup, create the bucket if that's false. And on the API call, just ignore the variable.